wandb/apis/public/api.py

```diff
@@ -1172,14 +1161,31 @@ def artifact(self, name, type=None):
             # set entity to match the settings since in above code it was potentially set to an org
             entity = self.settings["entity"] or self.default_entity
         artifact = wandb.Artifact._from_name(
-            entity, project, artifact_name, self.client, organization
+            entity, project, artifact_name, self.client, organization, enable_tracking
         )
         if type is not None and artifact.type != type:
             raise ValueError(
                 f"type {type} specified but this artifact is of type {artifact.type}"
             )
         return artifact
 
+    @normalize_exceptions
+    def artifact(self, name, type=None):
+        """Return a single artifact by parsing path in the form `project/name` or `entity/project/name`.
+
+        Arguments:
+            name: (str) An artifact name. May be prefixed with project/ or entity/project/.
+                    If no entity is specified in the name, the Run or API setting's entity is used.
+                Valid names can be in the following forms:
+                    name:version
+                    name:alias
+            type: (str, optional) The type of artifact to fetch.
+
+        Returns:
+            A `Artifact` object.
+        """
```

Comment:
you could also add a section about the error it raises. see this example for reference: https://google.github.io/styleguide/pyguide.html#383-functions-and-methods